agent/agent.go

```diff
@@ -158,7 +158,11 @@ LOOP:
 			}
 			// if the host's current task group is finished we teardown
 			if nextTask.ShouldTeardownGroup {
-				needPostGroup = nextTask.ShouldTeardownGroup
+				a.runPostGroupCommands(ctx, tc)
+				needPostGroup = false
+				// Running the post group commands implies exiting the group, so
+				// destroy prior task information.
+				tc = &taskContext{}
```

Comment:
I think this means that teardown group doesn't get logged anywhere useful.

Reply:
I'm not sure what you mean by this.